# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/api/FileSystemFileHandle.cpp
# Commit: 214c4d6bf7f9
# Full Hash: 214c4d6bf7f98a332c4ab81d325d9245ae810955
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1792245
# File Overlap Count: 2
# Description:
#   Bug 1792245 - Add ErrorResult to FileSystemRequestHandler methods; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159572
# ==============================================================================

diff -r 330c69218a93 -r 214c4d6bf7f9 dom/fs/api/FileSystemFileHandle.cpp
--- a/dom/fs/api/FileSystemFileHandle.cpp	Thu Oct 20 07:33:41 2022 +0300
+++ b/dom/fs/api/FileSystemFileHandle.cpp	Thu Oct 20 04:21:30 2022 +0000
@@ -53,7 +53,10 @@
     return nullptr;
   }
 
-  mRequestHandler->GetFile(mManager, mMetadata, promise);
+  mRequestHandler->GetFile(mManager, mMetadata, promise, aError);
+  if (aError.Failed()) {
+    return nullptr;
+  }
 
   return promise.forget();
 }
@@ -77,7 +80,10 @@
     return nullptr;
   }
 
-  mRequestHandler->GetAccessHandle(mManager, mMetadata, promise);
+  mRequestHandler->GetAccessHandle(mManager, mMetadata, promise, aError);
+  if (aError.Failed()) {
+    return nullptr;
+  }
 
   return promise.forget();
 }